frontend/dockerfile/instructions/commands_runmount.go

```diff
@@ -11,25 +11,33 @@ import (
 	"github.com/pkg/errors"
 )
 
-const MountTypeBind = "bind"
-const MountTypeCache = "cache"
-const MountTypeTmpfs = "tmpfs"
-const MountTypeSecret = "secret"
-const MountTypeSSH = "ssh"
+type MountType = string
```

Comment:
nit: I think we want

```suggestion
type MountType string
```

The `=` defines a type alias, which are meant for refactoring code across packages.

Reply:
So, this was mostly intentional. I'm using aliases (see them as "pseudo" "types") for these as throughout the code these are _actually_ just strings. The consts in the code are mostly used as a "utility" for convenience, but no strong type enforcement is currently happening.

I picked aliases to make the change less disruptive. These values are used for parsing strings passed as input, and switching to use a strong type would really complicate those, e.g. the below would require quite a refactor;

```go
func parseVal(val string) {
    switch strings.Tolower(val) {
    case FooEnabled:
        // something
    case BarEnabled:
        // something else
    default:
        panic("unknown option")
    }
}
```

The "pseudo types" _do_ still bring some of the benefits;

- Generated docs shows them nicely together
- Intent is clearer (these are all the `MountType` options)
- Intent on fields and arguments (`setMountType(t MountType)`), but still allowing to pass a regular string (without casting); validation is already handled for these, so invalid strings would (_should_) already be handled.
- Same for fields (`struct Foo{Field: MountType}`)
- IDE assistance (click the `MountType` above to navigate to all `MountType` options); same in docs (click the field navigates to the type definition, and the consts defined)

That said, we _could_ switch to using a strong type at some point (if we see a strong need for that).

Reply:
I'd agree with @aaronlehmann here - I think we should have a strong preference for creating a new type, consistent with other places where we have enums: `UsageRecordType`, `UlimitName`, `CapID`, etc. The only place where we properly use alias types is for the shared `Result` generic across different packages.

I don't think using a new type should increase the commit footprint significantly - though even if it does, I think it's worth the consistency with the rest of the codebase.